packages/client/hmi-client/src/services/workflow.ts

```diff
@@ -584,6 +585,41 @@ export class WorkflowWrapper {
 			downstreamNodes: outputEdges.map((e) => e.target && cache.get(e.target)).filter(Boolean) as WorkflowNode<any>[]
 		};
 	};
+
+	runDagreLayout() {
+		const g = new dagre.graphlib.Graph({ compound: true });
+		g.setGraph({});
+		g.setDefaultEdgeLabel(() => ({}));
+		g.graph().rankDir = 'LR';
+		this.getNodes().forEach((node) => {
+			g.setNode(node.id, {
+				label: node.displayName,
+				width: node.width ?? 50,
+				height: node.height ?? 50
```

Comment:
A couple of minor things
- I would remove the `width/height ?? 50` thing cause we should always have the width/height
- Looking at the video, you may want to play it bit with `g.graph().nodesep` or `g.graph().ranksep` to space things out a bit more, another way is to artificially increase the width/height by some padding amount

Reply:
Thanks, I've added the suggestions